lib/action_policy/lookup_chain.rb

```diff
@@ -53,21 +53,18 @@ def call(record, **opts)
 
       private
 
-      def lookup_within_namespace(policy_name, namespace)
-        return unless namespace
-        NamespaceCache.fetch(namespace.name, policy_name) do
+      def lookup_within_namespace(policy_name, namespace, strict_namespace = false)
+        cached_policy = NamespaceCache.fetch(namespace.name, policy_name) do
```

Comment:
I think, cache should take into account the strict option. Otherwise we may end up with the wrong value in the cache.

Let's add a test that verifies that caches distinguish strict vs. non-strict lookup.

Reply:
That should work the following way:

- `strict: true` — cache the result with the strict modifier (for the cache key)
- `strict: false` and policy is found for the provided namespace — add two cache values for both strict and non-strict lookup
- `strict: false` and policy is found for the outer namespace (or not found) — cache only for non-strict lookup.

Let me try to describe it with code:

```ruby
module A
  class BPolicy
  class APolicy


  module B
    class BPolicy
  end
end

# cache miss
lookup(:a, namespace: A::B) #=> A::APolicy

# cache hit
lookup(:a, namespace: A::B)  #=> A::APolicy

# cache miss
lookup(:b, namespace: A::B) #=> A::B::BPolicy

# (!!!) cache hit
lookup(:b, namespace: A::B, strict_namespace: true) #=> A::B::BPolicy

# cache miss
lookup(:b, namespace: A, strict_namespace: true) #=> A::BPolicy

# (!!!) cache hit
lookup(:b, namespace: A) #=> A::BPolicy
```